.github/workflows/release.yml

```diff
@@ -42,71 +43,53 @@ jobs:
-          RELEASE_CERT_MANAGER: ${{ steps.release.outputs.klt-cert-manager--release_created }}
-          RELEASE_METRICS_OPERATOR: ${{ steps.release.outputs.metrics-operator--release_created }}
+          CHANGED_ITEMS: ${{ steps.release.outputs.paths_released }}
           KLT_TAG: ${{ steps.release.outputs.tag_name }}
-          CERT_MANAGER_TAG: ${{ steps.release.outputs.klt-cert-manager--tag_name }}
-          METRICS_OPERATOR_TAG: ${{ steps.release.outputs.metrics-operator--tag_name }}
+          ARTIFACT_TAG_NAME: ${{ steps.release.outputs[format('{0}--tag_name', steps.release.outputs.paths_released[0])] }}
         with:
           script: |
-            const { RELEASE_KLT, RELEASE_CERT_MANAGER, RELEASE_METRICS_OPERATOR, KLT_TAG, CERT_MANAGER_TAG, METRICS_OPERATOR_TAG } = process.env
-
-            var result = []
-            if (RELEASE_KLT === "true") {
-              result.push(...[
-                {
-                    name: "lifecycle-operator",
-                    folder: "lifecycle-operator/",
-                    tagName: KLT_TAG
-                },
-                {
-                    name: "scheduler",
-                    folder: "scheduler/",
-                    tagName: KLT_TAG
-                },
-                {
-                    name: "functions-runtime",
-                    folder: "functions-runtime/",
-                    tagName: KLT_TAG
-                },
-                {
-                    name: "python-runtime",
-                    folder: "python-runtime/",
-                    tagName: KLT_TAG
+            const changedItems = JSON.parse(process.env.CHANGED_ITEMS || '[]');
+            console.log("changed items", changedItems);
+            const eligibleItems = JSON.parse(process.env.PUBLISHABLE_ITEMS || '[]');
+            console.log("eligible items", eligibleItems);
+            const itemsToPublish = changedItems.filter(i => eligibleItems.includes(i));
+            console.log("items to publish", itemsToPublish);
+
+            console.log("setting up build matrix...");
+            const releaseMatrix = itemsToPublish.map(item => {
+                switch (item) {
+                    case "lifecycle-operator":
+                        return [
+                            {
+                                name: item,
+                                folder: item,
+                                tagName: process.env.ARTIFACT_TAG_NAME
+                            },
+                            {
+                                name: "scheduler",
+                                folder: "scheduler/",
+                                tagName: process.env.ARTIFACT_TAG_NAME
+                            }
+                        ];
+                    case "metrics-operator":
+                    case "klt-cert-manager":
+                    case "runtimes/deno-runtime":
+                    case "runtimes/python-runtime":
+                        return [{
```

Comment:
default cases where just the artifact itself needs to be built.
lifecycle-operator is different cus it needs to be bundled together with scheduler and the base klt artifact `.` doesn't have any artifacts anymore since everything is its own thing now in the monorepo